domain/storageprovisioning/state/filesystem.go

```diff
@@ -0,0 +1,325 @@
+) (map[string]life.Life, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+	return st.getFilesystemLifeForNetNode(ctx, db, netNodeUUID)
+}
+
+// getFilesystemLifeForNetNode returns a mapping of filesystem ids to current
+// life value for each machine provisioned filesystem that is to be
+// provisioned by the machine owning the supplied net node.
+func (st *State) getFilesystemLifeForNetNode(
+	ctx context.Context,
+	db domain.TxnRunner,
+	netNodeUUID domainnetwork.NetNodeUUID,
+) (map[string]life.Life, error) {
+	netNodeInput := netNodeUUIDRef{UUID: netNodeUUID.String()}
+	stmt, err := st.Prepare(`
+SELECT DISTINCT (sf.filesystem_id, sf.life_id) AS (&filesystemLife.*)
+FROM            storage_filesystem sf
+JOIN            storage_filesystem_attachment sfa ON sf.uuid=sfa.storage_filesystem_uuid
+WHERE           sf.provision_scope_id=1
+AND             sfa.net_node_uuid=$netNodeUUIDRef.net_node_uuid
+		`, filesystemLife{}, netNodeInput)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+	var fsLives filesystemLives
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		exists, err := st.checkNetNodeExists(ctx, tx, netNodeUUID)
+		if err != nil {
+			return err
+		} else if !exists {
+			return errors.Errorf("net node %q does not exist", netNodeUUID)
+		}
+		err = tx.Query(ctx, stmt, netNodeInput).GetAll(&fsLives)
+		if err != nil && !errors.Is(err, sql.ErrNoRows) {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+	return maps.Collect(fsLives.Iter), nil
+}
+
+// InitialWatchStatementMachineProvisionedFilesystems returns both the
+// namespace for watching filesystem life changes where the filesystem is
+// machine provisioned. On top of this the initial query for getting all
+// filesystems in the model that are machine provisioned is returned.
+//
+// Only filesystems that can be provisioned by the machine connected to the
+// supplied net node will be emitted.
+func (st *State) InitialWatchStatementMachineProvisionedFilesystems(
+	netNodeUUID domainnetwork.NetNodeUUID,
+) (string, eventsource.Query[map[string]life.Life]) {
+	query := func(ctx context.Context, db database.TxnRunner) (
+		map[string]life.Life, error,
+	) {
+		return st.getFilesystemLifeForNetNode(ctx, db, netNodeUUID)
+	}
+	return "storage_filesystem_life_machine_provisioning", query
+}
+
+// InitialWatchStatementModelProvisionedFilesystems returns both the namespace
+// for watching filesystem life changes where the filesystem is model
+// provisioned. On top of this the initial query for getting all filesystems
+// in the model that model provisioned is returned.
+func (st *State) InitialWatchStatementModelProvisionedFilesystems() (string, eventsource.NamespaceQuery) {
+	query := func(ctx context.Context, db database.TxnRunner) ([]string, error) {
+		stmt, err := st.Prepare(`
+SELECT &filesystemID.*
+FROM storage_filesystem
+WHERE provision_scope_id=0
+`,
```

Comment:
```
SELECT &filesystemID.*
FROM   storage_filesystem
WHERE  provision_scope_id=0
```

Reply:
Can probably just fix this in a drive-by. Let's land it.